activeweb/src/main/java/org/javalite/activeweb/SessionFacade.java

```diff
@@ -134,102 +134,142 @@ public void setTimeToLive(int seconds){
     }
 
+    /**
+     * Add object to a session.
+     * @param key name of object
+     * @param value object reference.
+     */
     @Override
-    public Object get(Object key) {
-        return get(key.toString());
-    }
-
-    @Override
-    public Object put(Object key, Object value) {
-        Object val = get(key.toString());
-        put(key.toString(), (Serializable)value);
-        return val;
+    public Object put(String key, Object value) {
+        Object prev = get(key);
+        RequestContext.getHttpRequest().getSession(true).setAttribute(key, value);
+        return prev;
     }
 
     @Override
     public Object remove(Object key) {
-        Object val = get(key.toString());
-        RequestContext.getHttpRequest().getSession(true).removeAttribute(key.toString());
-        return val;
+        HttpSession session = httpSession();
+        if (session != null) {
+            Object val = session.getAttribute(key.toString());
+            session.removeAttribute(key.toString());
+            return val;
+        }
+        return null;
     }
 
     @Override
     public void putAll(Map m) {
-        Set keys = m.keySet();
-        for(Object k:keys){
-            put(k, m.get(k));
+        HttpSession session = RequestContext.getHttpRequest().getSession(true);
+        for (Object o : m.entrySet()) {
+            Map.Entry entry = (Entry) o;
+            session.setAttribute(entry.getKey().toString(), entry.getValue());
         }
     }
 
     @Override
     public void clear() {
-        throw new UnsupportedOperationException();
+        HttpSession session = httpSession();
+        if (session != null) {
+            for(String name : keySet()) {
+                session.removeAttribute(name);
+            }
+        }
     }
 
     @Override
-    public Set<Object> keySet() {
-        Set<Object> keys = new HashSet<>();
-        Enumeration names = RequestContext.getHttpRequest().getSession(true).getAttributeNames();
-        while (names.hasMoreElements()){
-            Object name = names.nextElement();
-            keys.add(name);
+    public Set<String> keySet() {
+        HttpSession session = httpSession();
+        if (session != null) {
+            Set<String> keys = new HashSet<>();
+            Enumeration<String> names = session.getAttributeNames();
+            while (names.hasMoreElements()) {
+                keys.add(names.nextElement());
+            }
+            return keys;
         }
-        return keys;
+        return Collections.emptySet();
     }
 
     @Override
-    public Collection values() {
-        Set<Object> values = new HashSet<>();
-        Enumeration names = RequestContext.getHttpRequest().getSession(true).getAttributeNames();
-        while (names.hasMoreElements()){
-            Object name = names.nextElement();
-            values.add(get(name));
+    public Collection<Object> values() {
+        HttpSession session = httpSession();
+        if (session != null) {
```

Comment:
Use `exists()`

Reply:
We can lost the session between two requests to RequestContext. I think this version is more safe.